Mathlib/Analysis/NormedSpace/Banach.lean

```diff
@@ -353,6 +353,30 @@ theorem coeFn_toContinuousLinearEquivOfContinuous_symm (e : E ≃ₗ[𝕜] F) (h
 
 end LinearEquiv
 
+namespace ContinuousLinearMap
+
+variable [CompleteSpace E]
+
+/-- An injective continuous linear map with a closed range defines a continuous linear equivalence
+between its domain and its range. -/
+noncomputable def rangeEquiv (f : E →L[𝕜] F) (hinj : Injective f) (hclo : IsClosed (range f)) :
+    E ≃L[𝕜] LinearMap.range f :=
+  have : CompleteSpace (LinearMap.range f) := hclo.completeSpace_coe
+  LinearEquiv.toContinuousLinearEquivOfContinuous (LinearEquiv.ofInjective f.toLinearMap hinj) <|
+    (f.continuous.codRestrict fun x ↦ LinearMap.mem_range_self f x).congr fun _ ↦ rfl
+
+@[simp]
+theorem rangeEquiv_apply (f : E →L[𝕜] F) (hinj : Injective f) (hclo : IsClosed (range f)) (x : E) :
+    f.rangeEquiv hinj hclo x = f x :=
```

Comment:
Please use equality of functions here (unapplied) and call it `coe_rangeEquiv`

Reply:
I've replaced this lemma with `(f.rangeEquiv hinj hclo : E → LinearMap.range f) = f.rangeRestrict`, so that when `rangeEquiv` is applied as a function, `simp` can use all the lemmas for `ContinuousLinearMap.rangeRestrict`. I will wait for your confirmation before merging this one.

Reply:
No, I think we want to go straight to `f`, but now that you mention it, we're missing the `simp` lemma `coe_rangeRestrict` as well, so please add that.

Reply:
Going straight to `f` has the issue that `f` maps to `F` but `f.rangeEquiv _ _` maps to `LinearMap.range f`. Then the equality between functions will not work unless I insert a composition with inclusion.

Reply:
I've added `coe_rangeRestrict`, but I'm not understanding your suggestion. Could you write the lemma statement of going straight to `f`?

Reply:
Sorry, I was talking nonsense. Ignore me.